internal/devbox/devbox.go

```diff
@@ -819,23 +804,34 @@ func (d *Devbox) ensureStateIsUpToDateAndComputeEnv(
 ) (map[string]string, error) {
 	defer debug.FunctionTimer().End()
 
-	// When ensureStateIsUpToDate is called with ensure=true, it always
-	// returns early if the lockfile is up to date. So we don't need to check here
-	if err := d.ensureStateIsUpToDate(ctx, ensure); isConnectionError(err) {
-		if !fileutil.Exists(d.nixPrintDevEnvCachePath()) {
-			ux.Ferrorf(
+	if envOpts.RecomputeEnv.Disabled {
+		upToDate, _ := d.lockfile.IsUpToDateAndInstalled(isFishShell())
+		if !upToDate {
+			ux.FHidableWarning(
```

Comment:
This won't work. Hidable warnings need the format string to match. You are using a non format string as a format string (which is why govet is complaining). That means this string won't match the `HideMessage` call earlier on.

One way to fix:

Have `StateOutOfDateMessage` be only the part that differs. So something like:


```golang
ux.FHidableWarning(
	ctx,
	d.stderr,
	StateOutOfDateMessage, 
        envOpts.RecomputeEnv.StateOutOfDateMessage,
)
```

Not a blocker, but I this feels a bit fragile and connects the cobra command to the devbox code in a funky way. Maybe one way to improve this in the future would be to use lifecycle hooks:

```golang
envOpts.Lifecycle.OnEnvNotUpToDate = func() { ux.Warning } 
```

Reply:
Yeah, this feels very icky. I like the lifecycles hooks approach. 👍🏾